java/src/main/java/lesson05/solution/Publisher.java

```diff
@@ -0,0 +1,54 @@
+package lesson05.solution;
+
+import com.google.common.collect.ImmutableMap;
+import io.dropwizard.Application;
+import io.dropwizard.Configuration;
+import io.dropwizard.setup.Environment;
+import io.jaegertracing.internal.JaegerTracer;
+import io.opentracing.Scope;
+import io.opentracing.Tracer;
+import lib.Tracing;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+
+public class Publisher extends Application<Configuration> {
+
+    private final Tracer tracer;
+
+    private Publisher(Tracer tracer) {
+        this.tracer = tracer;
+    }
+
+    @Path("/publish")
+    @Produces(MediaType.TEXT_PLAIN)
+    public class PublisherResource {
+
+        @GET
+        public String format(@QueryParam("helloStr") String helloStr, @Context HttpHeaders httpHeaders) {
+            try (Scope scope = Tracing.startServerSpan(tracer, httpHeaders, "publish")) {
```

Comment:
If you  really want to implement the fill solution for the extra credit, then this should also use the opentracing-contrib library for Dropwizard. The point of extracredit is to not do instrumentation manually.